consensus/enclave/impl/src/lib.rs

```diff
@@ -1127,6 +1143,49 @@ mod tests {
         );
     }
 
+    #[test_with_logger]
+    fn test_enclave_init_refuses_nested_multisig_governors_when_not_supported(logger: Logger) {
```

Comment:
:thought_balloon: This is sufficient for testing that `enclave_init()` obeys the governor validation logic, but it feels like the governor map validation logic itself could be more robustly tested.  This goes back to the suggestion of moving the validation to it's own function where one can focus on `BlockChainConfig`.
Some boundary cases:
- Empty map (this is exercised but it's not intentional and hard to locate in the tests)
- map with more than one item
  - all flat
  - First with multi
  - a middle entry has multi
  - the last element multi

Especially if the validation logic is moved to it's own function these cases should be able to be concise.
Though these test cases seem trivial and can be easily said to work by looking at the code, the intent is to provide guard rails for future maintainers.  For instance the `mint_trust_root_public_key` could get refactored to use the method from the Enclave structure (I would suggest updating to a freestanding function).  Without robust tests around this trivial logic (there is currently a test) someone can't refactor with confidence that they aren't inadvertently breaking something.

Reply:
Hm, the map here is a hashmap of `TokenId` -> `SignerSet` so not sure what you mean by first with multi, middle entry with multi, etc. The `SignerSet` has two separate lists - one for individual pubkeys and one for `SignerSet`s.

Reply:
How do you feel about the [revised tests](https://github.com/mobilecoinfoundation/mobilecoin/blob/6d02f913c2532715f206e8fb91441d51654b0701/consensus/enclave/api/src/config.rs#L318-L493)?

Reply:
Ping @nick-mobilecoin

Reply:
No problem! Thanks for the reviews.
I am not seeing any new nits though...